src/hugr/hugrmut.rs

```diff
@@ -267,6 +274,36 @@ where
         let cur = self.as_mut().op_types.get_mut(node.index);
         std::mem::replace(cur, op.into())
     }
+
+    fn insert_hugr(&mut self, root: Node, other: &impl AsPortgraph) -> Result<Node, HugrError> {
+        let node_map = self.as_mut().graph.insert_graph(other.as_portgraph())?;
+        let other_root = node_map[&other.root().index];
+
+        // Update hierarchy and optypes
+        self.as_mut().hierarchy.push_child(other_root, root.index)?;
+        for (&node, &new_node) in node_map.iter() {
+            other
+                .children(node.into())
+                .try_for_each(|child| -> Result<(), HugrError> {
+                    self.as_mut()
+                        .hierarchy
+                        .push_child(node_map[&child.index], new_node)?;
+                    Ok(())
+                })?;
+            let optype = other.get_optype(node.into());
+            self.as_mut().op_types.set(new_node, optype.clone());
```

Comment:
would it be more useful to have a way of moving the subgraph in to the bigger one rather than copying?

Reply:
Uhm, yeah.
Moving the graph and hierarchy contents is not that useful because they are mostly node pointers, but for the actual weights we should be able to `move` them.

Reply:
do you want that to be a separate method or update  this one?

Reply:
I think we'll need to split
```rust
fn insert_hugr(&mut self, root: Node, other: AsMut<Hugr>) -> ...; // or just `other: Hugr`
fn insert_hugr_view(&mut self, root: Node, other: &impl HugrView) -> ...;
```
If we just update this one we won't be able to insert filtered regions (or other read-only views).

Reply:
i think `other: Hugr` is fine no? for `insert_hugr`